dask/array/tests/test_gufunc.py

```diff
@@ -541,3 +541,49 @@ def mysum(x, res):
     y = mysum(a, axis=0, keepdims=True, allow_rechunk=True)
 
     assert_eq(x, y)
+
+
+def run_mangled_output_dtype_test(reassign_vectorize_dtypes=None):
+    def identity(x):
+        return x
+
+    o_dtype = np.dtype('U2')
+
+    if reassign_vectorize_dtypes is None:
+        f = da.gufunc(identity,
+                      output_dtypes=(o_dtype,),
+                      signature="()->()",
+                      vectorize=True)
+    else:
+        f = da.gufunc(identity,
+                      output_dtypes=(o_dtype,),
+                      signature="()->()",
+                      vectorize=True,
+                      reassign_vectorize_dtypes=reassign_vectorize_dtypes)
```

Comment:
It would be good to test with a few different dtypes and signatures (perhaps one outputting multiple arrays).